src/desktop.c

```diff
@@ -139,6 +140,56 @@ desktop_cycle_view(struct server *server, struct view *start_view,
 	return iter(&server->views, start_view, criteria);
 }
 
+/*
+ *  is either box overlapping the other by 80% of its own size?
+ */
+static int
+is_mostly_overlapping(struct wlr_box *a, struct wlr_box *b)
+{
+	int x_overlap = MAX(0, MIN(a->x + a->width, b->x + b->width)
+				- MAX(a->x, b->x));
+	int y_overlap = MAX(0, MIN(a->y + a->height, b->y + b->height)
+				- MAX(a->y, b->y));
+	int overlap_area = x_overlap * y_overlap;
+	int area_a = a->width * a->height;
+	int area_b = b->width * b->height;
+	return overlap_area > 0.80 * area_a || overlap_area > 0.80 * area_b;
+}
+
+struct view *
+desktop_cycle_view_near_region(struct server *server, struct view *active_view,
+		struct region *region)
+{
+	assert(region);
+
+	struct view *(*iter)(struct wl_list *head, struct view *view,
+		enum lab_view_criteria criteria);
+	bool forwards = true;
+
+	/*
+	 * if the currently active/focused window is mostly within the region,
+	 * then we want to cycle through in reverse order so that we
+	 * cycle through all windows, and not just the two most recent.
+	 */
+	if (active_view && is_mostly_overlapping(&active_view->current, &region->geo)) {
+		forwards = false;
+	}
+	iter = forwards ? view_next_no_head_stop : view_prev_no_head_stop;
+
+	enum lab_view_criteria criteria = rc.window_switcher.criteria;
+
+	struct view *cur;
+	cur = iter(&server->views, active_view, criteria);
+	while (cur && cur != active_view) {
+		if (!cur->minimized
+			&& is_mostly_overlapping(&cur->current, &region->geo)) {
```

Comment:
CodeStyle nitpick: double intend for condition continuation lines.
E.g. something like
```c
		if (!cur->minimized && is_mostly_overlapping(
				&cur->current, &region->geo)) {
			return cur;
		}
```
Or just keep everything in a single line as long as its < 100 cols.